### ISSUE TITLE
Execute-MSI fails on some workstations with 'Specified Cast Is Not Valid'

### ISSUE THREAD

**Issue by Kittzus** [PSAppDeployToolkit/PSAppDeployToolkit#84]
Using PSADT 3.6.5 and the following command line:

Execute-MSI -Action Install -Path 'googlechromestandaloneenterprise.msi'

and I'm getting the below error on some hosts. Any further information you need to look into this, please just let me know.
```
[Installation] :: Failed to resolve registry path [HKLM:SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall]. 
Error Record:
-------------

Message        : Specified cast is not valid.
InnerException : 

FullyQualifiedErrorId : System.InvalidCastException,Microsoft.PowerShell.Comman                ds.GetItemPropertyCommand
ScriptStackTrace      : bei <ScriptBlock>, C:\Windows\ccmcache\6y\AppDeployTool
                        kit\AppDeployToolkitMain.ps1: Zeile 1913
                        bei Get-InstalledApplication<Process>, C:\Windows\ccmca
                        che\6y\AppDeployToolkit\AppDeployToolkitMain.ps1: 
                        Zeile 1913
                        bei Execute-MSI<Process>, C:\Windows\ccmcache\6y\AppDep
                        loyToolkit\AppDeployToolkitMain.ps1: Zeile 2279
                        bei <ScriptBlock>, 
                        C:\Windows\ccmcache\6y\Deploy-Application.ps1: Zeile 
                        180
                        bei <ScriptBlock>, <Keine Datei>: Zeile 1
                        bei <ScriptBlock>, <Keine Datei>: Zeile 1

PositionMessage : In C:\Windows\ccmcache\6y\AppDeployToolkit\AppDeployToolkitMa
                  in.ps1:1913 Zeichen:165
                  +                     [psobject[]]$regKeyApplication = 
                  Get-ChildItem -LiteralPath $regKey -ErrorA ...
                  +    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

**Comment by Kittzus** [comment]
Just to clarify, it may well be that the host has an underlying issue with PS here, but the problem is that ADT seems to interpret this failure as "Application Already Installed" and skips its Install phase, meaning the post-installation configuration occurs anyway and clients report as "Application not detected after installation" in SCCM 2012.

**Comment by mmashwani** [comment]
Can you manually run this command on that machine and see what you get?
Get-ChildItem -LiteralPath 'HKLM:SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall' -ErrorAction 'SilentlyContinue' -ErrorVariable '+ErrorUninstallKeyPath'

Also run this and see what you get?
Get-ChildItem -LiteralPath 'HKLM:SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall' -ErrorAction 'Stop' -ErrorVariable '+ErrorUninstallKeyPath'

**Comment by mmashwani** [comment]
Also, could invalid data in the registry be your problem?:
https://connect.microsoft.com/PowerShell/feedbackdetail/view/801007/get-itemproperty-specified-cast-is-not-valid-error-when-reg-dword-value-has-binary-data

**Comment by mmashwani** [comment]
Suppressing errors stemming from invalid data in the registry and then continue the processing of the registry key was kind of tricky. I had to get rid of the Try/Catch and also redirect the error stream by using the 2>$null notation.
https://github.com/PSAppDeployToolkit/PSAppDeployToolkit/commit/88c279511de8d8a6ea894497da5994fa6ab9578d

I committed the fix to the the latest 3.6.6 beta. Test it out and let me know how it works out:
https://github.com/PSAppDeployToolkit/PSAppDeployToolkit/tree/master/Toolkit/AppDeployToolkit
